Shared/Samples/Densify and generalize geometry/DensifyAndGeneralizeGeometryView.swift

```diff
@@ -0,0 +1,188 @@
+        @Published var maxDeviation = 10.0
+        
+        /// A Boolean indicating whether to densify.
+        @Published var shouldDensify = false
+        
+        /// The max segment length for densifying.
+        @Published var maxSegmentLength = 100.0
+        
+        init() {
+            originalPolyline = Polyline(points: pointCollection)
+            
+            // Create map.
+            map = Map(basemapStyle: .arcGISStreetsNight)
+            
+            // Set the initial viewpoint to show the extent of the graphics.
+            map.initialViewpoint = Viewpoint(
+                center: originalPolyline.extent.center,
+                scale: 65907
+            )
+            
+            // Create graphics overlay.
+            let multipoint = Multipoint(points: pointCollection)
+            
+            // Create graphics for displaying the base points and lines.
+            let originalPolylineGraphic = Graphic(
+                geometry: originalPolyline,
+                symbol: SimpleLineSymbol(style: .dot, color: .red, width: 3)
+            )
+            let originalPointsGraphic = Graphic(
+                geometry: multipoint,
+                symbol: SimpleMarkerSymbol(style: .circle, color: .red, size: 7)
+            )
+            
+            // Add the graphics in the order we want them to appear, back to front.
+            graphicsOverlay = GraphicsOverlay(graphics: [
+                originalPointsGraphic,
+                originalPolylineGraphic,
+                resultPointsGraphic,
+                resultPolylineGraphic
+            ])
+        }
+        
+        /// Reset the model values to the originals.
+        func reset() {
+            shouldDensify = false
+            maxDeviation = 10.0
+            shouldGeneralize = false
+            maxSegmentLength = 100.0
+            updateGraphics()
+        }
+        
+        /// Update the result polyline and multipoint graphics after densifying
+        /// and generalizing.
+        func updateGraphics() {
+            // Reset the result graphics if there are no operations to do.
+            if !shouldGeneralize && !shouldDensify {
+                resultPolylineGraphic.geometry = nil
+                resultPointsGraphic.geometry = nil
+                return
+            }
+            
+            // Start with original polyline.
+            var resultPolyline = originalPolyline
+            
+            // Generalize the polyline with the specified max deviation.
+            if shouldGeneralize {
+                resultPolyline = GeometryEngine.generalize(
+                    resultPolyline,
+                    maxDeviation: maxDeviation,
+                    removeDegenerateParts: true
+                ) as! Polyline
+            }
+            // Densify the points of the polyline with the specified max segment
+            if shouldDensify {
+                resultPolyline = GeometryEngine.densify(
```

Comment:
not sure why this isn't generic in the Swift API. I'm assuming there is a reason, but I'd like to know why.  Same question for generalize(...) above.

Reply:
I am not totally sure.

Reply:
looks like it's because you can pass an envelope to these methods and get a polygon.